src/pybind/mgr/cephadm/module.py

```diff
@@ -1905,9 +1905,40 @@ def get_minimal_ceph_conf(self) -> str:
         })
         extra = self.extra_ceph_conf().conf
         if extra:
-            config += '\n\n' + extra.strip() + '\n'
+            try:
+                config = self._combine_confs(config, extra)
+            except Exception as e:
+                self.log.error(f'Failed to add extra ceph conf settings to minimal ceph conf: {e}')
         return config
 
+    def _combine_confs(self, conf1: str, conf2: str) -> str:
+        section_to_option: Dict[str, List[str]] = {}
+        final_conf: str = ''
+        for conf in [conf1, conf2]:
+            if not conf:
+                continue
+            section = ''
+            for line in conf.split('\n'):
+                if line.strip().startswith('#') or not line.strip():
```

Comment:
Do we really have to build our own config parser for this? I guess we have to, right? But is this the right place to start building a config parser in Python for ceph?

Reply:
Actually for now, minimal conf can't be created by user, so config parser is simply. The main purpose is to mix two confs. Any suggest?

Reply:
probably once we start refactoring cephadm (in reef by using the new packaged version) we can add/reuse support to parse conf and use it to perform this merging moving this code outside of cephadm.